packages/editor/src/components/entities-saved-states/hooks/use-is-dirty.js

```diff
@@ -18,38 +18,34 @@ const TRANSLATED_SITE_PROPERTIES = {
 };
 
 export const useIsDirty = () => {
-	const { dirtyEntityRecords } = useSelect( ( select ) => {
-		const dirtyRecords =
-			select( coreStore ).__experimentalGetDirtyEntityRecords();
+	const { editedEntities, siteEdits } = useSelect( ( select ) => {
+		const { __experimentalGetDirtyEntityRecords, getEntityRecordEdits } =
+			select( coreStore );
 
+		return {
+			editedEntities: __experimentalGetDirtyEntityRecords(),
+			siteEdits: getEntityRecordEdits( 'root', 'site' ),
+		};
+	}, [] );
+
+	const dirtyEntityRecords = useMemo( () => {
 		// Remove site object and decouple into its edited pieces.
-		const dirtyRecordsWithoutSite = dirtyRecords.filter(
+		const editedEntitiesWithoutSite = editedEntities.filter(
 			( record ) => ! ( record.kind === 'root' && record.name === 'site' )
 		);
 
-		const siteEdits = select( coreStore ).getEntityRecordEdits(
-			'root',
-			'site'
-		);
-
-		const siteEditsAsEntities = [];
+		const editedSiteEntities = [];
 		for ( const property in siteEdits ) {
-			siteEditsAsEntities.push( {
+			editedSiteEntities.push( {
 				kind: 'root',
 				name: 'site',
 				title: TRANSLATED_SITE_PROPERTIES[ property ] || property,
 				property,
 			} );
```

Comment:
Here it's interesting that we're looking only at the keys of `siteEdits`, not the values. So, when I'm editing the site title or tagline, then, as i type, the `siteEdits` change, and trigger a recalculation and rerender. But the keys don't change, only the value.

Fixing this would be a micro-optimization of a very minor use case (editing site properties), probably not worth doing. But interesting to know about 🙂

Reply:
So should we just return the record keys for this one? Do we have a data selector to ensure the same value on every render?

P.S. I've another minor refactoring in mind for this hook - using reducer for the `unselectedEntities` state. Happy to include this micro-optimization there if you want to pursue it.

Reply:
Now I found that the `__experimentalGetDirtyEntityRecords` selector has exactly the same problem. But this one is much more widely used. Although its return value is still the same, and although it's memoized with `createSelector`, the value is recalculated on every keystroke, when changing any block. Because the memoization is invalidated on every content value change, but the content values are not used to calculate the return value.

It won't be easy to memoize this properly.